.github/workflows/checkers-action.yml

```
@@ -0,0 +1,75 @@
+name: Checkers-Action
+
+on:
+  push:
+    branches: [ main ]
+    paths:
+      - 'cve_bin_tool/checkers/**/*.py'
+  pull_request:
+    branches: [ main ]
+    paths:
+      - 'cve_bin_tool/checkers/**/*.py'
+
+jobs:
+  run-script:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Get date 
+        id: get-date 
+        run: | 
+          echo "date=$(/bin/date -u "+%Y%m%d")" >> $GITHUB_OUTPUT 
+          echo "yesterday=$(/bin/date -d "-1 day" -u "+%Y%m%d")" >> $GITHUB_OUTPUT 
+
+      - name: Print Cache Keys 
+        run: | 
+          echo "Today's Cache Key: Linux-cve-bin-tool-${{ steps.get-date.outputs.date }}" 
+          echo "Yesterday's Cache Key: Linux-cve-bin-tool-${{ steps.get-date.outputs.yesterday }}" 
+
+      - name: Get today's cached database 
+        uses: actions/cache@13aacd865c20de90d75de3b17ebe84f7a17d57d2 # v4.0.0 
+        id: todays-cache 
+        with: 
+          path: cache 
+          key: Linux-cve-bin-tool-${{ steps.get-date.outputs.date }} 
+
+      - name: Get yesterday's cached database if today's is not available 
+        uses: actions/cache@13aacd865c20de90d75de3b17ebe84f7a17d57d2 # v4.0.0 
+        if: steps.todays-cache.outputs.cache-hit != 'true' 
+        with: 
+          path: cache 
+          key: Linux-cve-bin-tool-${{ steps.get-date.outputs.yesterday }}  
+
+      - name: Install cve-bin-tool
+        if: env.sbom != 'true'
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install --upgrade setuptools
+          python -m pip install --upgrade wheel
+          python -m pip install --upgrade -r dev-requirements.txt
+          python -m pip install --upgrade .
+          
```

I think you were given some bad advice above: since your script doesn't actually call anything out of cve-bin-tool you shouldn't need to install it or run it, merely copy the database to the expected location.   If you had any dependencies in your script that weren't python packages you could install them here (example below), but probably you just want to remove this section.  

```suggestion
      - name: Install requirements for vendor product pre-check
        run: |
          python -m pip install  YOUR_LIST_OF_REQUIREMENTS         
```